### PR TITLE
CTED-2353 | Use methods from cm_info instead of private properties.

### PR DESCRIPTION
**Pull request by MatthewRorke**

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -7293,7 +7293,7 @@ public function find_users($search) {
  * @param cm_info $cm Course-module object
  */
 function hsuforum_cm_info_view(cm_info $cm) {
-    if (!$cm->uservisible) {
+    if (!$cm->get_user_visible()) {
```
Comment (JJSwanevelder) [inline lib.php:7296]: @MatthewRorke - what is the difference between these two? Meaning the object attribute "uservisible", is that not always set hence the use for the function.

File: lib.php [hunk 2]
```diff
@@ -7307,7 +7307,7 @@ function hsuforum_cm_info_view(cm_info $cm) {
     }
 
     if ($unread = hsuforum_count_forum_unread_posts($cm, $cm->get_course())) {
-        $out .= '<a class="unread" href="' . $cm->url . '">';
+        $out .= '<a class="unread" href="' . $cm->get_url() . '">';
```
Comment (JJSwanevelder) [inline lib.php:7310]: Same question as on 7296

File: lib.php [hunk 3]
```diff
@@ -7316,7 +7316,7 @@ function hsuforum_cm_info_view(cm_info $cm) {
         $out .= '</a>';
     }
 
-    $cm->set_content($cm->content . $out); // append the unreadpost section to existing content
+    $cm->set_content($cm->get_formatted_content() . $out); // append the unreadpost section to existing content
```
Comment (JJSwanevelder) [inline lib.php:7319]: Are we 100% sure that the function "get_formatted_content" returns the same content as what should be in the content object attribute?

File: lib.php (hunk 1 above)
Comment (MatthewRorke) [reply lib.php:7296]: According to the `cm_info` class. all of the attributes used, such as uservisible, are private attributes. As seen in `lib/modinfolib.php`

File: lib.php (hunk 2 above)
Comment (MatthewRorke) [reply lib.php:7310]: The URL attribute is a private attribute and the function returns the private value. I believe this is part of a PSR standard.

File: lib.php (hunk 3 above)
Comment (MatthewRorke) [reply lib.php:7319]: The content attribute is private and in no way am I able to be 100% sure about this without deploying a test and getting someone to test, or testing it myself. However, this line is the only reason for all of the changes in this file as the $content attribute throws an issue in datadog due to it being undefined.

**Review by JJSwanevelder — APPROVED** [review]
👍

**Pull request merged**